gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/CheckExplicitSourceCompatibilityTask.java

```diff
@@ -127,4 +125,24 @@ public final void taskAction() throws IOException {
                 JavaVersion.current(),
                 getPath()));
     }
+
+    private JavaVersion getRawSourceCompat() {
+        // TODO(fwindheuser): Remove internal api usage. Maybe through adopting toolchains?
+        // We're doing this naughty casting because we need access to the `getRawSourceCompatibility` method.
+        if (GradleVersion.current().compareTo(GradleVersion.version("7.0")) < 0) {
+            org.gradle.api.plugins.internal.DefaultJavaPluginConvention convention =
+                    (org.gradle.api.plugins.internal.DefaultJavaPluginConvention)
+                            getProject().getConvention().getPlugin(JavaPluginConvention.class);
+            return convention.getRawSourceCompatibility();
+        }
+
+        DefaultJavaPluginExtension extension =
+                (DefaultJavaPluginExtension) getProject().getExtensions().getByType(JavaPluginExtension.class);
+        try {
+            Method rawSourceCompat = DefaultJavaPluginExtension.class.getMethod("getRawSourceCompatibility");
```

Comment:
Grim, but I can't immediately think of way to do this better (since the gradle jar we build against is 6.x, this API does not exist, right?). When this repo upgrades to 7, we'll probably need to switch this up and make the convention version use reflection, right?

Reply:
Yep, then we have to do reflection for the old method.